app/controllers/api/v1/meetings_controller.rb

```diff
@@ -0,0 +1,71 @@
+class Api::V1::MeetingsController < ApplicationController
+  before_action :set_users, only: [:create]
+  before_action :existing_partner_check, only: [:create]
+  before_action :set_meeting, only: [:update, :destroy]
+
+  def create
+    user = @users[:user]
+    partner = @users[:partner]
+
+    meeting = Meeting.create(meeting_params)
+
+    if meeting.save
+      UserMeeting.create(user: user, meeting: meeting, is_requestor: true)
+      UserMeeting.create(user: partner, meeting: meeting, is_requestor: false)
+      render json: MeetingSerializer.new(meeting), status: :created
+    else
+      render json: { error: meeting.errors.full_messages.to_sentence }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @meeting.nil?
+      render json: { error: 'Meeting not found.' }, status: :not_found
+    else
+      update_meeting_approval
+    end
+  end
+
+  def destroy
+    if @meeting
+      @meeting.user_meetings.destroy_all
+      @meeting.destroy
```

Comment:
If we have time to refactor, there's a way to do this on the models themselves, basically saying a user meeting is dependent on its meeting!